src/test/java/com/puppycrawl/tools/checkstyle/gui/TreeTableTest.java

```diff
@@ -108,4 +110,105 @@ public void testExpandOnEnterKey() {
+        xpathTextArea.setText("//IDENT");
+        findNodeButton.doClick();
+
+        final String expected = "/COMPILATION_UNIT/CLASS_DEF/IDENT"
+                + "[@text='InputTreeTableXpathAreaPanel']\n"
+                + "/COMPILATION_UNIT/PACKAGE_DEF/DOT/IDENT[@text='treetable']\n"
+                + "/COMPILATION_UNIT/PACKAGE_DEF/DOT[./IDENT[@text='treetable']]/DOT/IDENT"
+                + "[@text='gui']\n"
+                + "/COMPILATION_UNIT/CLASS_DEF[./IDENT[@text='InputTreeTableXpathAreaPanel']]"
+                + "/OBJBLOCK/CLASS_DEF/IDENT[@text='Inner']\n"
+                + "/COMPILATION_UNIT/PACKAGE_DEF/DOT[./IDENT[@text='treetable']]/DOT[./IDENT"
+                + "[@text='gui']]/DOT/IDENT[@text='checkstyle']\n"
+                + "/COMPILATION_UNIT/PACKAGE_DEF/DOT[./IDENT[@text='treetable']]/DOT[./IDENT"
+                + "[@text='gui']]/DOT[./IDENT[@text='checkstyle']]/DOT/IDENT[@text='tools']\n"
+                + "/COMPILATION_UNIT/PACKAGE_DEF/DOT[./IDENT[@text='treetable']]/DOT[./IDENT"
+                + "[@text='gui']]"
+                + "/DOT[./IDENT[@text='checkstyle']]/DOT[./IDENT[@text='tools']]/DOT/IDENT"
+                + "[@text='com']\n"
+                + "/COMPILATION_UNIT/PACKAGE_DEF/DOT[./IDENT[@text='treetable']]/DOT[./IDENT"
+                + "[@text='gui']]"
+                + "/DOT[./IDENT[@text='checkstyle']]/DOT[./IDENT[@text='tools']]/DOT[./IDENT"
+                + "[@text='com']]/IDENT[@text='puppycrawl']\n";
+
+        assertWithMessage("Expected and actual XPath queries should match.")
+                .that(xpathTextArea.getText())
+                .isEqualTo(expected);
+    }
+
+    @Test
+    public void testFindNodesMissingElements() throws IOException {
+        final MainFrame mainFrame = new MainFrame();
+        mainFrame.openFile(new File(getPath("InputTreeTableXpathAreaPanel.java")));
+        final JButton findNodeButton = findComponentByName(mainFrame, "findNodeButton");
+        final JTextArea xpathTextArea = findComponentByName(mainFrame, "xpathTextArea");
+        xpathTextArea.setText("//LITERAL_TRY");
+        findNodeButton.doClick();
+
+        final String expected = "No elements matching XPath query '//LITERAL_TRY' found.";
+
+        assertWithMessage("Unexpected XPath text area text")
+                .that(xpathTextArea.getText())
+                .isEqualTo(expected);
+    }
+
+    @Test
+    public void testFindNodesUnexpectedTokenAtStart() throws IOException {
+        final MainFrame mainFrame = new MainFrame();
+        mainFrame.openFile(new File(getPath("InputTreeTableXpathAreaPanel.java")));
+        final JButton findNodeButton = findComponentByName(mainFrame, "findNodeButton");
+        final JTextArea xpathTextArea = findComponentByName(mainFrame, "xpathTextArea");
+        xpathTextArea.setText("!*7^");
+        findNodeButton.doClick();
+
+        final String expected = "!*7^\n^ Unexpected token \"!\" at start of expression";
```

Comment:
Just curious, I assume the `^` on the next line is trying to line it up with the error character like saying "Here is the start of the issue".

Does this still look well if the error is further into the xpath?

Reply:
I carried this over from the original implementation. I do not think the carat is valuable. I have removed it from the original implementation in a separate commit, and removed it from my changes as well.

Done.